front/src/applications/editor/tools/rangeEdition/tool-factory.tsx

```diff
@@ -80,50 +82,72 @@ function getRangeEditionTool<T extends EditorRange>({
-        layersEntity.objType === 'SpeedSection'
-          ? {
-              id: 'reset-speed-section',
-              icon: BiReset,
-              labelTranslationKey: 'Editor.tools.speed-edition.actions.reset-speed-section',
-              isDisabled({ state: { entity, initialEntity } }) {
-                return isEqual(entity, initialEntity);
-              },
-              onClick({ setState, state: { initialEntity } }) {
-                setState({
-                  entity: cloneDeep(initialEntity),
-                });
-              },
-            }
-          : {
-              id: 'reset-catenary',
-              icon: BiReset,
-              labelTranslationKey: 'Editor.tools.catenary-edition.actions.reset-catenary',
-              isDisabled({ state: { entity, initialEntity } }) {
-                return isEqual(entity, initialEntity);
-              },
-              onClick({ setState, state: { initialEntity } }) {
-                setState({
-                  entity: cloneDeep(initialEntity),
-                });
-              },
-            },
+        {
+          id: `new-${getObjTypeAction(layersEntity.objType)}`,
+          icon: IoMdAddCircleOutline,
+          labelTranslationKey: `Editor.tools.${getObjTypeEdition(
+            layersEntity.objType
+          )}-edition.actions.new-${getObjTypeAction(layersEntity.objType)}`,
+          onClick({ setState }) {
+            setState(getInitialState());
+          },
+        },
+        {
+          id: `reset-${getObjTypeAction(layersEntity.objType)}`,
+          icon: BiReset,
+          labelTranslationKey: `Editor.tools.${getObjTypeEdition(
+            layersEntity.objType
+          )}-edition.actions.reset-${getObjTypeAction(layersEntity.objType)}`,
+          isDisabled({ state: { entity, initialEntity } }) {
+            return isEqual(entity, initialEntity);
+          },
+          onClick({ setState, state: { initialEntity } }) {
+            setState({
+              entity: cloneDeep(initialEntity),
+            });
+          },
+        },
+      ],
+      [
+        {
+          id: 'delete-entity',
+          icon: BsTrash,
+          labelTranslationKey: `Editor.tools.${getObjTypeEdition(
+            layersEntity.objType
+          )}-edition.actions.delete-${getObjTypeAction(layersEntity.objType)}`,
+          // Show button only if we are editing
+          isDisabled({ state }) {
+            return state.initialEntity.geometry.coordinates.length === 0;
+          },
+          onClick({ infraID, openModal, closeModal, forceRender, state, setState, dispatch, t }) {
+            openModal(
+              <ConfirmModal
+                title={t(
+                  `Editor.tools.${getObjTypeEdition(
+                    layersEntity.objType
+                  )}-edition.actions.delete-${getObjTypeAction(layersEntity.objType)}`
+                )}
+                onConfirm={async () => {
+                  await dispatch<ReturnType<typeof save>>(
```

Comment:
The await is not needed here
```suggestion
                  dispatch<ReturnType<typeof save>>(
```

Reply:
see above comment